bigbio/dataloader.py

```diff
@@ -0,0 +1,323 @@
+        exclude: Optional[list] = None,
+    ):
+        self.scripts = sorted(glob(os.path.join("biodatasets", "*", "*.py")))
+        self.tasks = tasks
+
+        # NOTE:
+        # my suggestion would be to force all LOCAL datasets
+        # to hard-code the upper folder name (where all files are contained)
+        # this way we can define a single `data_dir`
+        self.data_dir = data_dir
+
+        self.skip_huge = skip_huge
+        self.skip_weak = skip_weak
+        self.skip_local = skip_local
+        self.exclude = (
+            [Reference(s).to_name() for s in exclude] if exclude is not None else []
+        )
+
+    # pylint: disable=no-self-use
+    def load_module(self, ref: str) -> ModuleType:
+        """
+        Load dataloader script as python module
+        """
+
+        return importlib.import_module(Reference(ref).to_module())
+
+    def get_supported_tasks(self, module: Union[ModuleType, str]) -> list[Tasks]:
+        """
+        Wrapper to get _SUPPORTED_TASKS
+        """
+
+        if isinstance(module, str):
+            module = self.load_module(module)
+
+        return getattr(module, "_SUPPORTED_TASKS", [])
+
+    def _is_local(self, module: Union[ModuleType, str]) -> bool:
+        """
+        Wrapper to get _LOCAL
+        """
+
+        if isinstance(module, str):
+            module = self.load_module(module)
+
+        return getattr(module, "_LOCAL", False)
+
+    def _no_valid_tasks(self, tasks: list[Tasks]) -> bool:
+
+        if len(tasks) == 0:
+            return True
+
+        # skip dataset if tasks are specified
+        # and dataset does not have at least one selected supported task
+        return self.tasks is not None and not any(t in self.tasks for t in tasks)
+
+    def _skip_huge(self, script: str, config: str) -> bool:
+
+        name = Reference(script).to_name()
+
+        skip = bool(self.skip_huge and config in _HUGE.get(name, []))
+
+        if not self.skip_huge and skip:
+            logger.warning(
+                "The dataloader `{}` generates a HUGE dataset. Downloading and processing will take >5 minutes...",
+                name,
+            )
+
+        return skip
+
+    def _skip_weak(self, script: str, config: str) -> bool:
+
+        name = Reference(script).to_name()
+
+        skip = bool(self.skip_weak and config in _WEAK.get(name, []))
+
+        if not self.skip_huge and skip:
+            logger.warning(
+                "The dataloader `{}` generates a WEAK dataset. This is not recommended for training...",
+                name,
+            )
+
+        return skip
+
+    def get_by_task(self) -> dict:
+        """
+        Loads kwargs for `datasets.load_dataset` for all datasets according to tasks
+        """
+
+        bigbio_datasets = defaultdict(list)
+
+        for script in self.scripts:
+
+            if Reference(script).to_name() in self.exclude:
+                continue
+
+            module = self.load_module(script)
```

Comment:
the logic in `load_module` will break if this is run outside the package (e.g. try running from your home directory)